src/unused_finder/export_collector_tests.rs

```diff
@@ -0,0 +1,743 @@
+        let expected_map: HashSet<ExportedItem> = HashSet::new();
+        assert_eq!(expected_map, visitor.exported_ids);
+    }
+
+    #[test]
+    fn test_allowed_unused_export_const() {
+        let cm = Arc::<SourceMap>::default();
+        let comments = SingleThreadedComments::default();
+        let fm = cm.new_source_file(
+            FileName::Custom("test.ts".into()),
+            r#"
+                // @ALLOW-UNUSED-EXPORT
+                export const foo = 1;
+                "#
+            .to_string(),
+        );
+
+        let mut parser = create_test_parser(&fm, Some(&comments));
+
+        let module = parser.parse_typescript_module().unwrap();
+        let mut visitor = ExportsCollector::new(std::sync::Arc::new(vec![]), comments);
+        visit_module(&mut visitor, &module);
+        let expected_map: HashSet<ExportedItem> = HashSet::new();
+
+        assert_eq!(expected_map, visitor.exported_ids);
+    }
+
+    #[test]
+    fn test_allowed_unused_export_from() {
+        let cm = Arc::<SourceMap>::default();
+        let comments = SingleThreadedComments::default();
+        let fm = cm.new_source_file(
+            FileName::Custom("test.ts".into()),
+            r#"
+                // @ALLOW-UNUSED-EXPORT
+                export { foo } from './foo';
+                "#
+            .to_string(),
+        );
+
+        let mut parser = create_test_parser(&fm, Some(&comments));
+
+        let module = parser.parse_typescript_module().unwrap();
+        let mut visitor = ExportsCollector::new(std::sync::Arc::new(vec![]), comments);
+        visit_module(&mut visitor, &module);
+        let expected_map: HashMap<String, HashSet<ImportedItem>> = HashMap::new();
+        assert_eq!(expected_map, visitor.export_from_ids);
+    }
+
+    #[test]
+    fn test_allowed_unused_export_default_from() {
+        let cm = Arc::<SourceMap>::default();
+        let comments = SingleThreadedComments::default();
+        let fm = cm.new_source_file(
+            FileName::Custom("test.ts".into()),
+            r#"
+                // @ALLOW-UNUSED-EXPORT
+                export { default as foo } from './foo';
+                "#
+            .to_string(),
+        );
+
+        let mut parser = create_test_parser(&fm, Some(&comments));
+
+        let module = parser.parse_typescript_module().unwrap();
+        let mut visitor = ExportsCollector::new(std::sync::Arc::new(vec![]), comments);
+        visit_module(&mut visitor, &module);
+        let expected_map: HashMap<String, HashSet<ImportedItem>> = HashMap::new();
+        assert_eq!(expected_map, visitor.export_from_ids);
+    }
+
+    #[test]
+    fn test_allowed_unused_export_star_from() {
+        let cm = Arc::<SourceMap>::default();
+        let comments = SingleThreadedComments::default();
+        let fm = cm.new_source_file(
+            FileName::Custom("test.ts".into()),
+            r#"
+                // @ALLOW-UNUSED-EXPORT
+                export * from './foo';
```

Comment:
missing test guaranteeing that other exports in the same file are discovered when // @ALLOW-UNUSED-EXPORT